File: JANL/Types/DGVTemplate.vb
Imports System.IO
Imports System.Xml.Serialization

Namespace Types

	Public Class DGVTemplate

		Public Sub New()
			Columns = New List(Of DGVTemplateColumn)
		End Sub

		Public Property Columns As List(Of DGVTemplateColumn)

		Public Shared Function FromXML(XML As String) As DGVTemplate
			Dim XS As New XmlSerializer(GetType(DGVTemplate))
			Using SR = New StringReader(XML)
				Dim T = DirectCast(XS.Deserialize(SR), DGVTemplate)
				Return T
			End Using
		End Function

		Public Shared Function ToXML(T As DGVTemplate) As String
			Dim XS = New XmlSerializer(GetType(DGVTemplate))
			Using SW = New StringWriter()
				XS.Serialize(SW, T)
				Return SW.ToString()
			End Using
		End Function

	End Class
End Namespace

File: JANL.WinForms/Types/DGVTemplate.cs
using System.Collections.Generic;
using System.IO;
using System.Xml.Serialization;

namespace JANL.Types
{
    /// <summary>
    /// Шаблон <see cref="System.Windows.Forms.DataGridView"/>
    /// </summary>
    public class DGVTemplate
    {
        /// <summary>
        ///
        /// </summary>
        public DGVTemplate()
        {
            Columns = new List<DGVTemplateColumn>();
        }

        /// <summary>
        ///
        /// </summary>
        /// <param name="DGVColumns">Столбцы шаблона</param>
        public DGVTemplate(IEnumerable<DGVTemplateColumn> DGVColumns)
        {
            Columns = new List<DGVTemplateColumn>(DGVColumns);
        }

        /// <summary>
        /// Столбцы шаблона
        /// </summary>
        public List<DGVTemplateColumn> Columns { get; set; }

        /// <summary>
        /// Преобразует XML в шаблон
        /// </summary>
        /// <param name="xml">XML</param>
        /// <returns></returns>
        public static DGVTemplate FromXML(string xml)
        {
            XmlSerializer XS = new XmlSerializer(typeof(DGVTemplate));
            using (var SR = new StringReader(xml))
            {
                var T = (DGVTemplate)XS.Deserialize(SR);
                return T;
            }
        }

        /// <summary>
        /// Преобразует шаблон в XML
        /// </summary>
        /// <param name="template">Шаблон</param>
        /// <returns></returns>
        public static string ToXML(DGVTemplate template)
        {
            var XS = new XmlSerializer(typeof(DGVTemplate));
            using (var SW = new StringWriter())
            {
                XS.Serialize(SW, template);
                return SW.ToString();
            }
        }
    }
}